### PR TITLE
[TimeSeriesInsights] Query categorical variable and warm store query parameter changes

### PR DESCRIPTION
**Pull request by yeskarthik**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lirenhe** [comment]
@yeskarthik, please take a look at and fix the Avocado and spell check error.

**Comment by yeskarthik** [comment]
@lirenhe I fixed the comments, can you please complete the PR? Thanks.

**Comment by yeskarthik** [comment]
#sign-off

**Review by lirenhe — APPROVED** [review]

**Comment by yeskarthik** [comment]
@lirenhe can you please complete the PR?

**Comment by lirenhe** [comment]
> @lirenhe can you please complete the PR?

Please wait for ARM to signoff this PR

**Comment by yeskarthik** [comment]
@lirenhe this is not an ARM API, this is data access API, they usually don't sign off these changes. @deepakpalled

**Comment by yeskarthik** [comment]
I'm going to be added one more minor change.

**Review by lirenhe — APPROVED** [review]

**Comment by yeskarthik** [comment]
@Azure/azure-api-review-board can you please verify and merge these changes? These changes are already deployed and live. These are updates the existing APIs and only swagger changes.

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 1]
```diff
@@ -1261,6 +1339,76 @@
...
+      "properties": {
+        "label": {
+          "description": "The name of the category which will be used in constructing the output variable names.",
+          "type": "string"
+        },
+        "values": {
+          "description": "The list of values that a category maps to. Can be either a unique list of string or list of long.",
+          "type": "array",
+          "items": {
+            "type": "object"
```
Comment (johanste) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: The description and swagger are in disagreement as to what the types can be here. A `string` or `integer` in swagger/openapi are **not** `object`s.

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 2]
```diff
@@ -1261,6 +1339,76 @@
...
+              "description": "Categorical variable supports only 'step' interpolation.",
+              "$ref": "#/definitions/Interpolation"
+            },
+            "categories": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/TimeSeriesAggregateCategory"
+              }
+            },
+            "defaultCategory": {
```
Comment (johanste) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: Is there a specific reason as to why the type of `defaultCatgory` is a complex object with a single attribute?

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 3]
```diff
@@ -2334,6 +2480,14 @@
       "type": "string",
       "description": "Version of the API to be used with the client request. Currently supported version is \"2018-11-01-preview\"."
     },
+    "StoreType" : {
+      "name" : "storeType",
+      "in" : "query",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description":"For the environments with warm store enabled, the query can be executed either on the 'WarmStore' or 'ColdStore'. This parameter in the query defines which store the query should be executed on. If not defined, the query will be executed on the cold store."
```
Comment (johanste) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: Specify `default`?

**Pull request merged**